File: src/main/java/com/pixelreel/service/search/MovieSearchProvider.java
package com.pixelreel.service.search;

import com.pixelreel.dto.external.ExternalMovieSearchResult;
import com.pixelreel.dto.external.ProviderSearchResult;

public interface MovieSearchProvider {
  String id();

  ProviderSearchResult<ExternalMovieSearchResult> search(String query, int page);
}

File: express-backend/src/services/provider/MovieSearchProvider.ts
import { ProviderSearchResult, ExternalMovieSearchResult } from '../../dto/external-search';

// 影视搜索 Provider 接口，与 Java 端 MovieSearchProvider 对齐
export interface MovieSearchProvider {
  id(): string;
  search(query: string, page: number): Promise<ProviderSearchResult<ExternalMovieSearchResult>>;
}